packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts

```diff
@@ -0,0 +1,150 @@
+import {toHexString} from "@chainsafe/ssz";
+import {deneb, RootHex, ssz} from "@lodestar/types";
+import {ChainForkConfig} from "@lodestar/config";
+import {pruneSetToMax} from "@lodestar/utils";
+
+import {
+  BlockInputCacheType,
+  GossipedBlockInput,
+  BlockInput,
+  GossipedInputType,
+  getBlockInput,
+  BlockSource,
+  BlockInputBlobs,
+  BlobsCache,
+} from "../blocks/types.js";
+
+const MAX_GOSSIPINPUT_CACHE = 5;
+// ssz.deneb.BlobSidecars.elementType.fixedSize, 131256 is size for mainnet preset;
+const BLOBSIDECAR_FIXED_SIZE = ssz.deneb.BlobSidecars.elementType.fixedSize ?? 131256;
+
+export class SeenGossipBlockInput {
```

Comment:
it's great to have this as a separate cache, can you add unit test for this class? Also drop some comments for this class